src/app/AttributeCache.h

```diff
@@ -344,26 +364,46 @@ class AttributeCache : protected ReadClient::Callback
     void OnReportBegin() override;
     void OnReportEnd() override;
     void OnAttributeData(const ConcreteDataAttributePath & aPath, TLV::TLVReader * apData, const StatusIB & aStatus) override;
-    void OnError(CHIP_ERROR aError) override { return mCallback.OnError(aError); }
+    void OnError(CHIP_ERROR aError) override { mCallback.OnError(aError); }
 
     void OnEventData(const EventHeader & aEventHeader, TLV::TLVReader * apData, const StatusIB * apStatus) override
     {
-        return mCallback.OnEventData(aEventHeader, apData, apStatus);
+        mCallback.OnEventData(aEventHeader, apData, apStatus);
+    }
+
+    void OnDone() override
+    {
+        mRequestPathSet.clear();
+        return mCallback.OnDone();
     }
 
-    void OnDone() override { return mCallback.OnDone(); }
     void OnSubscriptionEstablished(uint64_t aSubscriptionId) override { mCallback.OnSubscriptionEstablished(aSubscriptionId); }
 
     void OnDeallocatePaths(chip::app::ReadPrepareParams && aReadPrepareParams) override
     {
-        return mCallback.OnDeallocatePaths(std::move(aReadPrepareParams));
+        mCallback.OnDeallocatePaths(std::move(aReadPrepareParams));
     }
 
+    virtual CHIP_ERROR OnUpdateDataVersionFilterList(DataVersionFilterIBs::Builder & aDataVersionFilterIBsBuilder,
+                                                     const Span<AttributePathParams> & aAttributePaths,
+                                                     bool & aEncodedDataVersionList) override;
+    virtual void OnReadingWildcardAttributePath(const AttributePathParams & aAttributePathParams) override;
+
+    // Committed the pending cluster data version, if there is one.
```

Comment:
```suggestion
    // Commit the pending cluster data version, if there is one.
```
Thought I had put that in last time, but clearly not....